s3fs/core.py

```diff
@@ -142,7 +169,10 @@ def _ls(self, path, refresh=False):
                     f['Size'] = 0
                     del f['Name']
             else:
-                files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                try:
+                    files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                except ClientError:
+                    files = []
```

Comment:
OK, I can go with your example - ls returns empty entries (size=0) to fill in the directory space, but not overwriting any entries that happen to be keys sharing the same name. 

Should they, then, be available to `open()` as empty files? (probably not)

Should they show up in walk or glob? (probably not)

Entries are already sorted in `_ls`.